lib/helpers.js

```diff
@@ -35,12 +36,22 @@ function openCache(options) {
 function fetchAndCache(request, options) {
   options = options || {};
   var successResponses = options.successResponses || globalOptions.successResponses;
-  return fetch(request.clone()).then(function(response) {
+  var maxCacheEntries = options.maxCacheEntries || globalOptions.maxCacheEntries;
+  var cacheName = options.cacheName || globalOptions.cacheName;
 
-    // Only cache GET requests with successful responses
+  return fetch(request.clone()).then(function(response) {
+    // Only cache GET requests with successful responses.
+    // Since this is not part of the promise chain, it will be done asynchronously and will not
+    // block the response from being returned to the page.
     if (request.method === 'GET' && successResponses.test(response.status)) {
-      openCache(options).then(function(cache) {
-        cache.put(request, response);
+      openCache({cacheName: cacheName}).then(function(cache) {
```

Comment:
I'm already doing the `cacheName = options.cacheName || globalOptions.cacheName` earlier in the code, so I figured it wasn't necessary to trigger that logic again inside the `openCache()` method. But I can restore it back to the original if you think it's cleaner.

(Refactoring all the options logic to use a helper like [`_.defaults()`](https://lodash.com/docs#defaults) might be nice in a future PR.)